tests/simple/testdata/comparisons.textproto

```diff
@@ -738,6 +746,61 @@ section {
     }
   }
 }
+section {
+  name: "in_list_literal"
+  description: "Set membership tests using list literals and the 'in' operator"
+  test {
+    name: "elem_not_in_empty_list"
+    expr: "'empty' in []"
+    value { bool_value: false }
+  }
+  test {
+    name: "elem_in_list"
+    expr: "'elem' in ['elem', 'el', 'em']"
+    value { bool_value: true }
+  }
+  test {
+    name: "elem_not_in_list"
+    expr: "'e' in ['elem', 'el', 'em']"
+    value { bool_value: false }
+  }
+  test {
+    name: "elem_in_mixed_type_list"
+    expr: "'el' in [1, 'el', 2]"
+    value { bool_value: true }
+  }
+  test {
+    name: "elem_in_mixed_type_list_error"
+    expr: "'em' in [1u, 'el', 2, b'em']"
+    eval_error: {
+      errors: { message: "no such overload" }
```

Comment:
This is super tricky. Unlike the map case, we do have a basic support for heterogeneous lists. The `elem_in_mixed_type_list` works b/c the element is there, but if you put `'xx'` instead of `'el'`, would you expect a `"no such overload"` error?

In the case of `elem_in_mixed_type_list_error` there is a string element, it's just not the right value.